### ISSUE TITLE
update logging

### ISSUE THREAD

**Issue by pohly** [intel/pmem-csi#645]
Several related tasks:
- update to klog v2 (https://groups.google.com/d/msgid/kubernetes-dev/CANw6fcFOLn4G1_e9f%2BNc9iwUDR%2BP7i%2BKt%2BnjDoXCdK1bPG8Mfw%40mail.gmail.com)
- start using the logr logger for structured logging in the operator
- make code take a logr instance through their context instead of using a global logger
- when running `go test ./pkg/pmem-csi-operator/controller/deployment/`, ensure that log output is printed through https://pkg.go.dev/testing?tab=doc#T.Logf (similar to https://github.com/intel/oim/blob/master/pkg/log/testlog/testlog.go) and thus visible only for failed tests

That last point depends on the previous ones. Output written through the global klog logger cannot be made test-specific because multiple tests run in parallel, hence we need to reduce its usage.

**Comment by pohly** [comment]
Some instructions for structured logging: https://github.com/kubernetes/community/pull/4793/files#diff-af969ddd1593b66ad7330155ce792feb

**Comment by pohly** [comment]
Let's tackle this for 0.8.x as part of making PMEM-CSI more admin-friendly.

**Comment by pohly** [comment]
Upstream hasn't decided on an approach for passing a logger via the context. It's also going to touch a lot of code in PMEM-CSI. Let's postpone to 0.9.

**Comment by pohly** [comment]
The `--logging-format` command line options and operator API change went into PR #825.

What remains is converting more log messages from plain text to structured output and (once ready) use the upstream klogr from https://github.com/kubernetes/klog/pull/197.

**Comment by pohly** [comment]
https://github.com/intel/pmem-csi/pull/870 replaced the forked klogr. I consider this issue resolved. Let's tackle converting log messages in pmem-csi-driver for 1.0 (https://github.com/intel/pmem-csi/issues/889).
